### ISSUE TITLE
Crash on Pixel 2 XL Oreo DP3

### ISSUE THREAD

**Issue by SimonMarquis** [JakeWharton/SdkSearch#108]
I'm getting a crash every time I do these steps:

1. Open SDK Search
2. Back (close Activity)
3. Re-open SDK Search
4. Input some text (crash at the first input)

Here is the full logcat of these steps:

```
    --------- beginning of main
07-04 21:34:52.469 811-811/? D/android.hardware.power@1.2-service.wahoo-libperfmgr: LAUNCH ON
07-04 21:34:52.475 26678-26678/? I/AppLifecycleTracker: onActivityPaused: 
07-04 21:34:52.490 29559-29559/? E/arton.sdksearc: Not starting debugger since process cannot load the jdwp agent.
07-04 21:34:52.492 1200-1248/? I/ActivityManager: Start proc 29559:com.jakewharton.sdksearch/u0a366 for activity com.jakewharton.sdksearch/.ui.MainActivity
07-04 21:34:52.516 29559-29559/? E/LoadedApk: Unable to instantiate appComponentFactory
    java.lang.ClassNotFoundException: Didn't find class "android.support.v4.app.CoreComponentFactory" on path: DexPathList[[zip file "/data/app/com.jakewharton.sdksearch-a2Dqq0qhe7KKRHtcZhBX1w==/base.apk", zip file "/data/app/com.jakewharton.sdksearch-a2Dqq0qhe7KKRHtcZhBX1w==/split_config.xxxhdpi.apk"],nativeLibraryDirectories=[/data/app/com.jakewharton.sdksearch-a2Dqq0qhe7KKRHtcZhBX1w==/lib/arm64, /system/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
        at android.app.LoadedApk.createAppFactory(LoadedApk.java:226)
        at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:731)
        at android.app.LoadedApk.getClassLoader(LoadedApk.java:810)
        at android.app.LoadedApk.getResources(LoadedApk.java:1032)
        at android.app.ContextImpl.createAppContext(ContextImpl.java:2345)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5749)
        at android.app.ActivityThread.access$1100(ActivityThread.java:199)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
    	Suppressed: java.io.IOException: No original dex files found for dex location /data/app/com.jakewharton.sdksearch-a2Dqq0qhe7KKRHtcZhBX1w==/split_config.xxxhdpi.apk
        at dalvik.system.DexFile.openDexFileNative(Native Method)
        at dalvik.system.DexFile.openDexFile(DexFile.java:354)
        at dalvik.system.DexFile.<init>(DexFile.java:101)
        at dalvik.system.DexFile.<init>(DexFile.java:75)
        at dalvik.system.DexPathList.loadDexFile(DexPathList.java:394)
        at dalvik.system.DexPathList.makeDexElements(DexPathList.java:354)
        at dalvik.system.DexPathList.<init>(DexPathList.java:164)
        at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:74)
        at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:65)
        at dalvik.system.PathClassLoader.<init>(PathClassLoader.java:64)
        at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:73)
        at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:88)
        at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:74)
        at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:40)
        at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:727)
        		... 12 more
07-04 21:34:52.516 29559-29559/? W/arton.sdksearc: JIT profile information will not be recorded: profile file does not exits.
07-04 21:34:52.540 29559-29577/? D/libEGL: loaded /vendor/lib64/egl/libEGL_adreno.so
07-04 21:34:52.549 29559-29577/? D/libEGL: loaded /vendor/lib64/egl/libGLESv1_CM_adreno.so
07-04 21:34:52.562 29559-29577/? D/libEGL: loaded /vendor/lib64/egl/libGLESv2_adreno.so
07-04 21:34:52.566 29559-29559/? D/NetworkSecurityConfig: No Network Security Config specified, using platform default
07-04 21:34:52.620 29559-29559/? D/OpenGLRenderer: Skia GL Pipeline
07-04 21:34:52.631 29559-29559/? I/SQLiteCompatibilityWalFlags: Read compatibility WAL flags: compatibility_wal_supported=true, wal_syncmode=NORMAL
07-04 21:34:52.659 29559-29582/? I/Adreno: QUALCOMM build                   : 7bf2852, Ie32bfa6f6f
    Build Date                       : 06/03/18
    OpenGL ES Shader Compiler Version: EV031.24.00.00
    Local Branch                     : googdrp
    Remote Branch                    : 
    Remote Branch                    : 
    Reconstruct Branch               : 
    Build Config                     : S L 4.0.10 AArch64
07-04 21:34:52.664 29559-29582/? I/Adreno: PFP: 0x005ff110, ME: 0x005ff066
07-04 21:34:52.668 29559-29582/? I/ConfigStore: android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasWideColorDisplay retrieved: 1
    android::hardware::configstore::V1_0::ISurfaceFlingerConfigs::hasHDRDisplay retrieved: 0
07-04 21:34:52.668 29559-29582/? I/OpenGLRenderer: Initialized EGL, version 1.4
07-04 21:34:52.668 29559-29582/? D/OpenGLRenderer: Swap behavior 1
07-04 21:34:52.699 1856-1856/? I/GoogleInputMethod: onFinishInput() : Dummy InputConnection bound
    onStartInput() : Dummy InputConnection bound
07-04 21:34:52.700 1200-1254/? I/ActivityManager: Displayed com.jakewharton.sdksearch/.ui.MainActivity: +217ms
07-04 21:34:52.709 811-811/? D/android.hardware.power@1.2-service.wahoo-libperfmgr: LAUNCH OFF
07-04 21:34:52.855 614-654/? W/SurfaceFlinger: Attempting to set client state on removed layer: Splash Screen com.jakewharton.sdksearch#0
    Attempting to destroy on removed layer: Splash Screen com.jakewharton.sdksearch#0
07-04 21:34:53.165 26678-26678/? I/Finsky: [2] com.google.android.finsky.ce.a.onTrimMemory(7): Memory trim requested to level 20
    [2] com.google.android.finsky.ce.a.onTrimMemory(13): Flushing in-memory image cache
07-04 21:34:53.168 26678-26678/? I/AppLifecycleTracker: onActivitySaveInstanceState: 
07-04 21:34:53.171 26678-26678/? I/AppLifecycleTracker: onActivityStopped: 
07-04 21:34:53.181 26678-26678/? V/MediaPlayer: resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
    cleanDrmObj: mDrmObj=null mDrmSessionId=null
07-04 21:34:53.301 1200-2742/? I/OpenGLRenderer: Initialized EGL, version 1.4
07-04 21:34:53.301 1200-2742/? D/OpenGLRenderer: Swap behavior 1
07-04 21:34:54.254 1856-1856/? I/GoogleInputMethod: onKeyDown() : keyCode = 4, event = KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0, eventTime=62651593, downTime=62651593, deviceId=-1, source=0x101 }
07-04 21:34:54.440 26678-26678/? I/AppLifecycleTracker: onActivityStarted: 
07-04 21:34:54.444 26678-26678/? I/AppLifecycleTracker: onActivityResumed: 
07-04 21:34:54.469 1856-1856/? I/GoogleInputMethod: onFinishInput() : Dummy InputConnection bound
07-04 21:34:54.470 1856-1856/? I/GoogleInputMethod: onStartInput() : Dummy InputConnection bound
07-04 21:34:55.126 614-655/? W/SurfaceFlinger: Attempting to destroy on removed layer: AppWindowToken{d56bd4e token=Token{d4f049 ActivityRecord{bce7650 u0 com.jakewharton.sdksearch/.ui.MainActivity t16402}}}#0
07-04 21:34:55.387 1200-3167/? I/ActivityManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=com.jakewharton.sdksearch cmp=com.jakewharton.sdksearch/.ui.MainActivity} from uid 10057
07-04 21:34:55.391 811-811/? D/android.hardware.power@1.2-service.wahoo-libperfmgr: LAUNCH ON
07-04 21:34:55.398 26678-26678/? I/AppLifecycleTracker: onActivityPaused: 
07-04 21:34:55.403 29559-29559/? W/ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@69a31b7
07-04 21:34:55.497 1200-1254/? I/ActivityManager: Displayed com.jakewharton.sdksearch/.ui.MainActivity: +96ms
07-04 21:34:55.499 1856-1856/? I/GoogleInputMethod: onFinishInput() : Dummy InputConnection bound
    onStartInput() : Dummy InputConnection bound
07-04 21:34:55.517 811-811/? D/android.hardware.power@1.2-service.wahoo-libperfmgr: LAUNCH OFF
07-04 21:34:55.682 614-2497/? W/SurfaceFlinger: Attempting to set client state on removed layer: Splash Screen com.jakewharton.sdksearch#0
    Attempting to destroy on removed layer: Splash Screen com.jakewharton.sdksearch#0
07-04 21:34:56.054 26678-26678/? I/Finsky: [2] com.google.android.finsky.ce.a.onTrimMemory(7): Memory trim requested to level 20
    [2] com.google.android.finsky.ce.a.onTrimMemory(13): Flushing in-memory image cache
07-04 21:34:56.056 26678-26678/? I/AppLifecycleTracker: onActivitySaveInstanceState: 
07-04 21:34:56.058 26678-26678/? I/AppLifecycleTracker: onActivityStopped: 
07-04 21:34:56.069 26678-26678/? V/MediaPlayer: resetDrmState:  mDrmInfo=null mDrmProvisioningThread=null mPrepareDrmInProgress=false mActiveDrmScheme=false
    cleanDrmObj: mDrmObj=null mDrmSessionId=null
07-04 21:34:56.197 1200-2742/? I/OpenGLRenderer: Initialized EGL, version 1.4
07-04 21:34:56.197 1200-2742/? D/OpenGLRenderer: Swap behavior 1
07-04 21:34:56.401 1856-1856/? I/GoogleInputMethod: onFinishInput() : Dummy InputConnection bound
07-04 21:34:56.413 29559-29559/? I/AssistStructure: Flattened final assist data: 1760 bytes, containing 1 windows, 5 views
07-04 21:34:56.420 1200-3167/? I/AssistStructure: Flattened final assist data: 1624 bytes, containing 1 windows, 5 views
07-04 21:34:56.430 1856-1856/? I/LatinIme: onActivate() : EditorInfo = Package = com.jakewharton.sdksearch : Type = Text : Learning = Disable : Suggestion = Hide : AutoCorrection = Disable : Microphone = Show : Incognito = Disable
    isMonolingualEnUs(): false. enable_autospace_after_punctuation_setting: true. pref_key_enable_autospace_after_punctuation: false. EditorInfoUtil.shouldEnableAutoCorrection(editorInfo): false.
07-04 21:34:56.431 1856-1856/? I/InputContext: input context reset success: true error: 0 input_state_id 72
07-04 21:34:56.437 1856-1856/? I/LatinMotionHandler: consumeEvent() : GestureTypingEnabled = true
07-04 21:34:56.438 1856-1856/? I/Conv2QueryExtension: isEnabled() : true
    Current locale: en_US, config allows these locales: de,en,fr,it,es,pt
    Conv2Query not enabled due to current app [com.jakewharton.sdksearch] not in whitelist
07-04 21:34:56.438 1856-1856/? W/Conv2QueryExtension: onActivate() : Disabled by unsupported host app
07-04 21:34:56.438 1856-1856/? I/EmojiDataExtension: onActivate() : Locale = en_US
07-04 21:34:56.438 1856-1856/? I/EmojiSuperpacksFeature: Emoji superpacks enabled
07-04 21:34:56.438 1856-1856/? I/FeatureCardNotice: Feature card notice not shown: no cards to display
07-04 21:34:56.438 1856-1856/? I/GifEntryExtension: onActivate()
07-04 21:34:56.438 1856-1856/? I/DevFeatureConfig: Gif candidate disabled by Phenotype
07-04 21:34:56.439 1856-1856/? W/VoiceImeTranscriptor: UIME feature enabled: true
07-04 21:34:56.439 1856-1856/? I/VoiceImeExtension: Voice Extension onStartInputView() Mic status = [MicIconHidden-SettingOff, ]
07-04 21:34:57.336 29559-29559/? D/AndroidRuntime: Shutting down VM
    
    
    --------- beginning of crash
07-04 21:34:57.336 29559-29559/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.jakewharton.sdksearch, PID: 29559
    kotlinx.coroutines.experimental.JobCancellationException: Child job was cancelled because of parent failure; job=StandaloneCoroutine{Cancelled}@af41eb2
    Caused by: kotlinx.coroutines.experimental.JobCancellationException: Job was cancelled normally; job=JobImpl{Cancelled}@d17c180
07-04 21:34:57.374 29559-29559/? I/Bugsnag: Saved unsent payload to disk (/data/user/0/com.jakewharton.sdksearch/cache/bugsnag-errors/1530732897337_startupcrash.json) 
07-04 21:34:57.374 29559-29593/? I/Bugsnag: Sending 1 saved error(s) to Bugsnag
07-04 21:34:57.376 1200-2017/? W/ActivityManager:   Force finishing activity com.jakewharton.sdksearch/.ui.MainActivity
07-04 21:34:57.379 1200-1247/? I/ActivityManager: Showing crash dialog for package com.jakewharton.sdksearch u0
07-04 21:34:57.386 1200-1246/? W/BroadcastQueue: Permission Denial: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to ProcessRecord{366e18a 29559:com.jakewharton.sdksearch/u0a366} (pid=29559, uid=10366) requires android.permission.READ_LOGS due to sender android (uid 1000)
07-04 21:34:57.577 29559-29593/? I/Bugsnag: Completed error API request
    Deleting sent error file 1530732897337_startupcrash.json
```

**Comment by JakeWharton** [comment]
Are you on 1.3.1 or 1.3.2?

**Comment by SimonMarquis** [comment]
1.3.2, fresh install from Play Store.

**Comment by JakeWharton** [comment]
Ok. Boy do I hate coroutine channels. Terrible for just about everything on Android.

**Comment by SimonMarquis** [comment]
:) Tell me if you need more traces or when you'll have a fix ready to test.

**Comment by JakeWharton** [comment]
No it's just a matter of adding a million logging calls to isolate where the problem actually is. I already fixed one occurrence in 1.3.2 of this so shouldn't be too hard to find the other...

**Comment by SimonMarquis** [comment]
When I had the previous version installed, I also experienced crashes when performing steps 1 to 3. This might be the one you already fixed.

**Comment by JakeWharton** [comment]
The crash I fixed occurred immediately upon launch the second time after back. You didn't even get to see the UI.

**Comment by JakeWharton** [comment]
Reproduced

**Comment by JakeWharton** [comment]
Fixed by 7917b814104f441670ffd5ddda6e18d1630e12eb

**Comment by JakeWharton** [comment]
Released as 1.3.3
